src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostsListPage.java

```diff
@@ -91,14 +94,17 @@ public PostsListPage open() {
   }
 
   public PostsListPage waitForPageReload() {
+    waitSafely(() -> wait.forElementVisible(By.className("loading-overlay")));
+    waitSafely(() -> wait.forElementNotVisible(By.className("loading-overlay")));
+    return this;
+  }
+
+  private void waitSafely(Runnable o) {
     try {
-      wait.forElementVisible(By.className("loading-overlay"));
+      o.run();
     } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)